pwnlib/util/misc.py

```diff
@@ -382,6 +385,27 @@ def run_in_new_terminal(command, terminal=None, args=None, kill_at_exit=True, pr
     if terminal == 'tmux':
         args += ['-F' '#{pane_pid}', '-P']
 
+    # if we're on a Mac and use iTerm
+    # we use `osascript` to split the current window
+    if terminal == 'osascript':
+        osa_script = """
+tell application "iTerm"
+    tell current session of current window
+        set newSession to (split horizontally with default profile)
+    end tell
+    tell newSession
+        write text "{gdb_command}"
```

Comment:
Does this need escaping if the command uses `"`?

Reply:
I don't think so. Since the `osa_script` is triple quoted, everything inside should be copied as is. So the quotes will be copied to the tmp file. Except if you think that the `command` could contain quotes

Reply:
do you think `write text "{gdb_command!r}"` be sufficient?

Reply:
Maybe reusing the code below which dumps the commands to a python script to avoid this question is the way to go. and then just run that python script in the osascript.

Reply:
@peace-maker thanks for your help. i just made this quick change as you suggested: i now use the previously created python script which will run the command. BUT if the command is a string we still have the same escaping issue.  
If we add another layer and try running the osascript from another python script, we would have the same result. I think... 🤔